src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py

```diff
@@ -299,3 +307,140 @@ def get_src_py_file_to_dest_py_file_map(
+                )
+                ddl_lst_per_ef.append(
+                    generate_grant_sql_ddl_statements(extension_function)
+                )
+
+            dest_file_py_file_to_ddl_map[py_file] = "\n".join(ddl_lst_per_ef)
+
+        return dest_file_py_file_to_ddl_map
+
+
+def generate_create_sql_ddl_statements(ex_fn: Dict[str, Any]) -> str:
+    """
+    Generates a "CREATE FUNCTION/PROCEDURE ... " SQL DDL statement based on a dictionary of extension function properties.
+    Logic for this create statement has been lifted from snowflake-snowpark-python v1.15.0 package.
+    """
+
+    object_type = ex_fn["object_type"]
+    object_name = ex_fn["object_name"]
+
+    if object_type == "PROCEDURE" and ex_fn.get("anonymous", False):
+        cc.warning(
+            dedent(
+                f"""{object_type.replace(' ', '-')} {object_name} cannot be an anonymous procedure in a Snowflake Native App.
+                    Skipping generation of 'CREATE FUNCTION/PROCEDURE ...' SQL statement for this object."""
+            )
+        )
+        return ""
+
+    replace_in_sql = f" OR REPLACE " if ex_fn["replace"] else ""
+
+    sql_func_args = ",".join(
+        [
+            f"{a['name']} {t}"
+            for a, t in zip(ex_fn["input_args"], ex_fn["input_sql_types"])
+        ]
+    )
+
+    imports_in_sql = (
+        f"\nIMPORTS=({ex_fn['all_imports']})" if ex_fn["all_imports"] else ""
+    )
+
+    packages_in_sql = (
+        f"\nPACKAGES=({ex_fn['all_packages']})" if ex_fn["all_packages"] else ""
+    )
+
+    external_access_integrations = ex_fn["external_access_integrations"]
+    external_access_integrations_in_sql = (
+        f"""\nEXTERNAL_ACCESS_INTEGRATIONS=({','.join(external_access_integrations)})"""
+        if external_access_integrations
+        else ""
+    )
+
+    secrets = ex_fn["secrets"]
+    secrets_in_sql = (
+        f"""\nSECRETS=({",".join([f"'{k}'={v}" for k, v in secrets.items()])})\n"""
+        if secrets
+        else ""
+    )
+
+    execute_as = ex_fn["execute_as"]
+    if execute_as is None:
+        execute_as_sql = ""
+    else:
+        execute_as_sql = f"\nEXECUTE AS {execute_as}"
+
+    inline_python_code = ex_fn["inline_python_code"]
+    if inline_python_code:
+        inline_python_code_in_sql = f"""\
+AS $$
+{inline_python_code}
+$$
+"""
+    else:
+        inline_python_code_in_sql = ""
+
+    try:
+        assert ex_fn["runtime_version"] is not None
+        assert ex_fn["handler"] is not None
+    except AssertionError as err:
+        cc.warning(f"Could not generate SQL DDL due to incorrect information:\n{err}")
+        return ""
+
+    create_query = dedent(
+        f"""\
+CREATE{replace_in_sql}
+{_get_object_type_as_text(object_type)} {'IF NOT EXISTS' if ex_fn["if_not_exists"] else ''}{object_name}({sql_func_args})
+{ex_fn["return_sql"]}
+LANGUAGE PYTHON
+RUNTIME_VERSION={ex_fn["runtime_version"]} {imports_in_sql}{packages_in_sql}{external_access_integrations_in_sql}{secrets_in_sql}
```

Comment:
just thought of this, but do we default the runtime version to the one from the sandbox executing the user's code? If not, we should add a task for that.

Reply:
Sounds like Snowpark is already doing what I thought it should, so that's OK for now. Currently, the runtime will always be populated by Snowpark then? If so, that's good enough.